skfda/exploratory/visualization/representation.py

```diff
@@ -0,0 +1,320 @@
+            if fig and ax are None.
+        n_cols(int, optional): designates the number of columns of the
+            figure to plot the different dimensions of the image. Only
+            specified if fig and ax are None.
+        n_points (int or tuple, optional): Number of points to evaluate in
+            the plot. In case of surfaces a tuple of length 2 can be pased
+            with the number of points to plot in each axis, otherwise the
+            same number of points will be used in the two axes. By default
+            in unidimensional plots will be used 501 points; in surfaces
+            will be used 30 points per axis, wich makes a grid with 900
+            points.
+        domain_range (tuple or list of tuples, optional): Range where the
+            function will be plotted. In objects with unidimensional domain
+            the domain range should be a tuple with the bounds of the
+            interval; in the case of surfaces a list with 2 tuples with
+            the ranges for each dimension. Default uses the domain range
+            of the functional object.
+        sample_labels (list of int): contains integers from [0 to number of
+            labels) indicating to which group each sample belongs to. Then,
+            the samples with the same label are plotted in the same color.
+            If None, the default value, each sample is plotted in the color
+            assigned by matplotlib.pyplot.rcParams['axes.prop_cycle'].
+        label_colors (list of colors): colors in which groups are
+            represented, there must be one for each group. If None, each
+            group is shown with distict colors in the "Greys" colormap.
+        label_names (list of str): name of each of the groups which appear
+            in a legend, there must be one for each one. Defaults to None
+            and the legend is not shown.
+        **kwargs: if dim_domain is 1, keyword arguments to be passed to
+            the matplotlib.pyplot.plot function; if dim_domain is 2,
+            keyword arguments to be passed to the
+            matplotlib.pyplot.plot_surface function.
+
+    Returns:
+        fig (figure object): figure object in which the graphs are plotted.
+
+    """
+
+    fig, axes = _get_figure_and_axes(chart, fig, axes)
+    fig, axes = _set_figure_layout_for_fdata(fdata, fig, axes, n_rows, n_cols)
+
+    if domain_range is None:
+        domain_range = fdata.domain_range
+    else:
+        domain_range = _list_of_arrays(domain_range)
+
+    sample_colors, patches = _get_color_info(
+        fdata, sample_labels, label_names, label_colors, kwargs)
+
+    if fdata.dim_domain == 1:
+
+        if n_points is None:
+            n_points = constants.N_POINTS_UNIDIMENSIONAL_PLOT_MESH
+
+        # Evaluates the object in a linspace
+        eval_points = np.linspace(*domain_range[0], n_points)
+        mat = fdata(eval_points, derivative=derivative, keepdims=True)
+
+        color_dict = {}
```

Comment:
Fixed in 0c800cacc0d587b691716c0278c0910f6ce916f2. Thank you for pointing this.

Reply:
Fine, although alias `c` still doesn't work, I don't know if intentionally.
```python
fd.plot(c='red')
```

Also, I thought it was possible to use a list with the colors of each sample. Maybe it could be include as a future enhancement.

```python
fd.plot(color=len(fd) * ['g'])
```

Reply:
> Fine, although alias `c` still doesn't work, I don't know if intentionally.
> 
> ```python
> fd.plot(c='red')
> ```

It was not intentional. Fixed in ae53434a36d29a4bc825ed0cc5e3775643f7d624.

> Also, I thought it was possible to use a list with the colors of each sample. Maybe it could be include as a future enhancement.
> 
> ```python
> fd.plot(color=len(fd) * ['g'])
> ```

This can be done combining `sample_labels` and `label_colors`.

Reply:
> It was not intentional. Fixed in ae53434.

Now works correctly.

> This can be done combining sample_labels and label_colors.

Then maybe the argument `color/c` and `sample_colors` have a similar functionality and can never be passed at the same time, but putting them together (not now) could simplify the api, but it's only an idea, maybe there are conflicts I haven't thought of.(I never remember the name of `sample_colors` and I have to go to the examples to look at it 😄 ).

Reply:
In fact it is not `sample_colors` but `sample_labels` and `label_colors` 😜. But I agree that the names are not very intuitive. I would change `sample_labels` to simply `labels`, but that could be confused with matplotlib `label` argument. We should try to improving the names in a future PR. Maybe `group`, `group_names` and `group_colors`?

Reply:
A proof that I never remember the name 😆 . 
I prefer these other names too, could be changed along with #165 in a future PR.